yarn-project/noir-contracts/src/contracts/private_token_airdrop_contract/src/storage.nr

```diff
@@ -1,24 +1,29 @@
-use dep::aztec::state_vars::{
-    map::Map,
-    set::Set
-};
-use dep::value_note::value_note::{
-    ValueNote,
-    ValueNoteMethods,
-    VALUE_NOTE_LEN,
-};
-use crate::claim_note::{CLAIM_NOTE_LEN, ClaimNote, ClaimNoteMethods};
+use crate::claim_note::{ClaimNote, ClaimNoteMethods, CLAIM_NOTE_LEN};
+use dep::aztec::context::{PrivateContext, PublicContext};
+use dep::aztec::state_vars::{map::Map, set::Set};
+use dep::std::option::Option;
+use dep::value_note::value_note::{ValueNote, ValueNoteMethods, VALUE_NOTE_LEN};
 
 struct Storage {
     balances: Map<Set<ValueNote, VALUE_NOTE_LEN>>,
     claims: Set<ClaimNote, CLAIM_NOTE_LEN>,
 }
 
 impl Storage {
-    fn init() -> Self {
+    fn init(
+        private_context: Option<&mut PrivateContext>,
+        public_context: Option<&mut PublicContext>,
+    ) -> Self {
         Storage {
-            balances: Map::new(1, |s| Set::new(s, ValueNoteMethods)),
-            claims: Set::new(2, ClaimNoteMethods),
+            balances: Map::new(
+                private_context,
+                public_context,
+                1, // Storage slot
+                |private_context, public_context, slot| {
+                    Set::new(private_context, public_context, slot, ValueNoteMethods)
+                },
+            ),
+            claims: Set::new(private_context, public_context, 2, ClaimNoteMethods),
```

Comment:
I hadn't appreciated how ugly storage declarations would become with this new change. Declaring storage like this doesn't make for a very nice user experience, does it? (I know I initially proposed the change, to clean up calls to methods (by removing the `context` as a parameter to those calls)... but this is an ugly trade off, and I'm not sure if it's a good one, in hindsight).
I returned to this PR after a breaking change was flagged on discord @benesjan @spalladino - this PR caused a breaking change to storage declarations, and someone was struggling with declaring a double-mapping with this new syntax (and I can understand why :) )

Reply:
Declaring a double mapping has become this monstrosity, haha:

```rust
Storage {
    double_mapping: Map::new(
        private_context,
        public_context,
        1,
        |private_context, public_context, slot1| Map::new(
            private_context,
            public_context,
            slot1,
            |private_context, public_context, slot2| Set::new(
                private_context,
                public_context,
                slot2,
                ValueNoteMethods
            )
        )
    )
}
```

This might have to be classified as "unacceptable syntax"?

Reply:
I don't think they'll be implemented for some weeks (at the earliest). Perhaps we could wrap the `private_context, public_context` "tuple" in a `context` type, just to reduce the verbosity of this.

```
struct Context {
    private_context: Option<PrivateContext>,
    public_context: Option<PublicContext>,
    _is_private: bool
}
```
And then pass this ugly thing between state variables instead?

A double mapping is a nice idea too!

Reply:
> TLDR: Can you push the Noir team to get traits implemented soon? :-P

I just asked, for you:
<img width="236" alt="image" src="https://github.com/AztecProtocol/aztec-packages/assets/42943676/1e2604b1-a0e3-4e24-8962-193ac09198ce">

Reply:
> Perhaps we could wrap the private_context, public_context "tuple" in a Context type, just to reduce the verbosity of this.

I think that sould be easy to do (annoying, but easy) until we do get traits. Do you want to create an issue and assign it to Otters?